### ISSUE TITLE
Token in example module is being trimmed inccorectly

### ISSUE THREAD

**Issue by amitaibu** [Gizra/message#85]
`admin/structure/message/manage/example_create_node`

![edit_message_type___site-install-2](https://cloud.githubusercontent.com/assets/125707/14065447/446ced94-f432-11e5-8677-147312bb993d.jpg)

**Comment by HelenaEksler** [comment]
@amitaibu 
Seems like a `ckeditor` issue. Happens with any token in the `href` attribute. 

![selection_031](https://cloud.githubusercontent.com/assets/8306902/15384058/562260c2-1da0-11e6-8f15-0230901df89b.png)
![selection_032](https://cloud.githubusercontent.com/assets/8306902/15384060/578f6d10-1da0-11e6-90f0-8cdf05b572c3.png)

**Comment by HelenaEksler** [comment]
Same for regular body field:
![selection_033](https://cloud.githubusercontent.com/assets/8306902/15384232/958d6148-1da1-11e6-8f82-5ed4ae846ae1.png)

Will try to find the solution.

**Comment by ribel** [comment]
Hi guys!
I have investigated this issue a little bit and looks like it's not CKEditor.
As i see we didn't store text format, and that's cause this issue.
The easiest, way is to hardcode `'#format' => 'full_html'` in `MessageTypeMultipleTextField::singleElement()`, but probably need better solution. 

![image](https://cloud.githubusercontent.com/assets/2241917/16773547/0c016d08-4861-11e6-8304-c0421ff7bb50.png)

**Comment by amitaibu** [comment]
Thanks @ribel , we would love to get a PR with a (non hardocded full format) fix ;)

**Comment by pcambra** [comment]
I think we're hitting a core bug, not related with CKEditor but haven't found anything in the core issue queue, so I'd appreciate a second pair of eyes.

Steps to test this: 
- Enable message & message example modules.
- Select a filter that has "Limit allowed HTML tags and correct faulty HTML" selected (admin/config/content/formats/manage). I've created one that has just that option.
- Access the message example module create node screen: admin/structure/message/manage/example_create_node
- You will see something like this (what is reported in #0) https://monosnap.com/file/HtSDdTOOr5NosSJtkVtN9kBU3NQ9NH.png which I think it means is not javascript related.

Internally it goes to:
-  \Drupal\editor\EditorXssFilter\filterXss (string is OK)
- \Drupal\Component\Utility\Xss:filter (string is OK)
- \Drupal\Component\Utility\Xss:split (string is OK)
- \Drupal\Component\Utility\Xss:attributes (string is OK)
- \Drupal\Component\Utility\UrlHelper:filterBadProtocol (string is OK)
- \Drupal\Component\Utility\UrlHelper:stripDangerousProtocols <- here is where we lose it

In `$protocol = substr($uri, 0, $colonpos);`, it considers [message: a protocol and it strips it out.

I'm going to check again in the issue queues and create a core bug for this if needed.

**Comment by pcambra** [comment]
Probably related issues:

https://www.drupal.org/node/2544110
https://www.drupal.org/node/2567743
https://www.drupal.org/node/2635632

Edit: Added another related issue

**Comment by amitaibu** [comment]
@pcambra thanks for investigating it!

**Comment by catch56** [comment]
I'd normally expect token replacement to run before a text format.

**Comment by amitaibu** [comment]
Yes, that indeed makes more sense. @pcambra care to continue here? :)

**Comment by pcambra** [comment]
I might be misunderstanding the solution, but in this case it's an example, we don't have the message to replace the tokens, they are "template" tokens.

**Comment by amitaibu** [comment]
Message can use two kinds of tokens. One, provided by core's token module, and it's being replaced [here](https://github.com/Gizra/message/blob/22e2a0fb93565459117c23a307ca457f0aa2a3b3/src/Entity/Message.php#L317). The other, is by providing Message hardcoded replacement values.

In this case we are using token module tokens. 

Maybe we should indeed change the term to `replacement values` or something like this, to prevent confusion.

**Comment by pcambra** [comment]
As far as I understand this issue, we're in this url: admin/structure/message/manage/example_create_node and we're talking about tokens such as message:user-url, message:user-name, message:node-url and message:node-title. All of those are token module based tokens.

If we run token replacements before the text format kicks in, I think we need a message entity to get the user and node from. At least for the example_create_node (and related) screen.

If we can't replace them, not sure how to show the message:user-url otherwise without being trimmed out by Xss filters (without trying to change core behaviour for tokens)

**Comment by amitaibu** [comment]
Sorry, I thing I was also confused, as the trimming happens before any token replacement. The input filter simply "eats" the text.

**Comment by jhedstrom** [comment]
In #128, token replacement still happens _after_ text formatting, so this will still be an issue. If it would resolve this, what about just pulling text formatting out of `MessageTemplate::getText()` and processing it instead in `Message::getText()`, which is where arguments and tokens are processed too?

**Comment by amitaibu** [comment]
The problem in this issue is that tokens are being "eaten", but in fact no token replacement should happen at this stage -- this is just the message edit form.

**Comment by jhedstrom** [comment]
I'm not sure this is something we can fix. It might need to be fixed in core itself... either way, I don't think it need block beta, so I'm removing this from that milestone. It's still tagged as blocking a full release, but we can revisit that when we get there.

**Comment by jhedstrom** [comment]
Here's the core issue: https://www.drupal.org/node/2372127
